code/modules/power/cell.dm

```diff
@@ -167,7 +167,7 @@
 
 /obj/item/weapon/stock_parts/cell/pulse //80 pulse shots
 	name = "pulse rifle power cell"
-	maxcharge = 16000
+	maxcharge = 4000

```

Comment:
I would leave it like this :laughing:
Beyond joking though it shouldn't be infinite otherwise you will make them unkillable by firepower.